.github/workflows/ci-build.yaml

```diff
@@ -401,7 +401,15 @@ jobs:
       - run: yarn --cwd ui test
       - run: yarn --cwd ui lint
       - run: yarn --cwd ui deduplicate
-      - run: git diff --exit-code
+      - if: github.actor == 'dependabot[bot]'
```

Comment:
I was thinking of adding this to the [`dependabot-reviewer`](https://github.com/argoproj/argo-workflows/blob/feaed7ac8cf8a0667fc20b8be5bd590bef4da42a/.github/workflows/dependabot-reviewer.yml) Workflow instead, so all dependabot things are in one place. That may be slightly more complicated though.

Reply:
This whole block should also have a comment describing it and why it exists

Reply:
In order for this job to work in dependabot-reviewer, I think it is necessary to run change-files like ci-build workflow to determine whether the ui job is running.
But I think that creates too much complexity.

Reply:
Mmm not so much `changed-files` as we'd need to run parts of the `ui:` job itself in order to install Node, NPM, and some deps. It would indeed be simpler to leave here 🤔